contracts/modules/cover/Cover.sol

```diff
@@ -240,29 +250,32 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
 
     // convert to NXM amount
     uint nxmPriceInCoverAsset = pool().getTokenPrice(params.coverAsset);
-    uint remainderAmountInNXM = 0;
+    uint requestedAmountInNXM = 0;
     uint totalCoverAmountInNXM = 0;
-
     uint _coverSegmentsCount = _coverSegments[coverId].length;
 
     for (uint i = 0; i < allocationRequests.length; i++) {
 
-      require(allocationRequests[i].coverAmountInAsset > 0, "Cover: coverAmountInAsset = 0");
+      uint coverAmountInAsset = allocationRequests[i].coverAmountInAsset;
+      require(coverAmountInAsset > 0, "Cover: coverAmountInAsset = 0");
 
-      uint requestedCoverAmountInNXM
-        = allocationRequests[i].coverAmountInAsset * NXM_IN_WEI / nxmPriceInCoverAsset + remainderAmountInNXM;
+      // adding capacity unallocated in the last round to the one requested for the current one
+      requestedAmountInNXM += Math.divCeil(coverAmountInAsset * NXM_IN_WEI, nxmPriceInCoverAsset);
```

Comment:
reminder

We discussed dropping this functionality and just reverting if can't allocate as much as you wanted.


Would simplify the logic a lot.

Reply:
Handled in a new commit, please check it out